tests/e2e/csi_snapshot_basic.go

```diff
@@ -1109,26 +1124,50 @@ var _ = ginkgo.Describe("Volume Snapshot Basic Test", func() {
 		gomega.Expect(isDiskAttached2).To(gomega.BeTrue(), "Volume is not attached to the node")
 
 		ginkgo.By("Verify the volume is accessible and Read/write is possible")
-		cmd := []string{"exec", pod.Name, "--namespace=" + namespace, "--", "/bin/sh", "-c",
-			"cat /mnt/volume1/Pod1.html "}
+		var cmd []string
+		if windowsEnv {
```

Comment:
There is lot of redundancy of this below logic where we do `if windows do this else do that.`
but we are calling the same method of createPod. Readability can be improved with the below logic

One solution can be to use a map. Look at his below logic
```
var envCmdMap = map[string]string{
    "windows": windowscmd,
    "linux":   linuxcmd,
} 
```
In createPod - Variable ` envCmdMap` can be passed to the method.

Reply:
+1

Reply:
removed redundancy using some utility functions